lib/sitemap.d.ts

```diff
@@ -0,0 +1,145 @@
+import builder = require('xmlbuilder');
+import SitemapItem = require('./sitemap-item');
+/**
+ * Shortcut for `new Sitemap (...)`.
+ *
+ * @param   {Object}        conf
+ * @param   {String}        conf.hostname
+ * @param   {String|Array}  conf.urls
+ * @param   {Number}        conf.cacheTime
+ * @param   {String}        conf.xslUrl
+ * @param   {String}        conf.xmlNs
+ * @return  {Sitemap}
+ */
+export declare function createSitemap(conf: {
+    urls: string | Sitemap["urls"];
+    hostname: string;
+    cacheTime: number;
+    xslUrl: string;
+    xmlNs?: string;
+}): Sitemap;
+export declare class Sitemap {
+    limit: number;
+    hostname: string;
+    urls: (string | {
+        url: string;
+        root?: Sitemap["root"];
+        img?: any;
+        links?: {
+            url: string;
+        }[];
+    })[];
+    cacheResetPeriod: number;
+    cache: string;
+    xslUrl: string;
+    xmlNs: string;
+    root: builder.XMLElementOrXMLNode & {
+        attributes?: [];
+        children?: [];
+        instructionBefore?(...argv: any[]): any;
+    };
+    cacheSetTimestamp: number;
+    /**
+     * Sitemap constructor
+     * @param {String|Array}  urls
+     * @param {String}        hostname    optional
+     * @param {Number}        cacheTime   optional in milliseconds; 0 - cache disabled
+     * @param {String}        xslUrl            optional
+     * @param {String}        xmlNs            optional
+     */
+    constructor(urls: string | Sitemap["urls"], hostname: string, cacheTime: number, xslUrl: string, xmlNs: string);
+    /**
+     *  Clear sitemap cache
+     */
+    clearCache(): void;
+    /**
+     *  Can cache be used
+     */
+    isCacheValid(): boolean;
+    /**
+     *  Fill cache
+     */
+    setCache(newCache: any): string;
+    /**
+     *  Add url to sitemap
+     *  @param {String} url
+     */
+    add(url: any): number;
+    /**
+     *  Delete url from sitemap
+     *  @param {String} url
+     */
+    del(url: any): number;
+    /**
+     *  Create sitemap xml
+     *  @param {Function}     callback  Callback function with one argument — xml
+     */
+    toXML(callback: any): string;
```

Comment:
Using this you are missing the types for the callback, i.e. if your callback will have 2 arguments you can do something like this:

```
export type Callback<E, T> = (err: E, data: T) => void;

toXML(callback: Callback<Error, string>): void;
```


```suggestion
    toXML(callback: Callback<Error, string>): void;
    toXML(): string;
```